osquery/core/watcher.cpp

```diff
@@ -50,7 +48,17 @@ FLAG(int32,
      1,
      "Performance limit level (0=loose, 1=normal, 2=restrictive, 3=debug)");
 
-FLAG(bool, disable_watchdog, false, "Disable userland watchdog process");
+CLI_FLAG(bool, disable_watchdog, false, "Disable userland watchdog process");
+
+/// If the worker exits the watcher will inspect the return code.
+void childHandler(int signum) {
+  siginfo_t info;
+  waitid(P_ALL, 0, &info, WEXITED | WSTOPPED | WNOHANG | WNOWAIT);
```

Comment:
add a note here as to why WNOWAIT is used (since there is a watcher thread requesting the queued wait information).